### PR TITLE
[new featrue] add cryptor plugin

### PR DESCRIPTION
**Pull request by impactCn**
// Describe your PR here; eg. Fixes #issueNo

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor/).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-agent/shenyu-agent-bootstrap/dependency-reduced-pom.xml [hunk 1]
```diff
@@ -0,0 +1,143 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (yu199195) [inline shenyu-agent/shenyu-agent-bootstrap/dependency-reduced-pom.xml:2]: remove this

File: shenyu-agent/shenyu-agent-core/dependency-reduced-pom.xml [hunk 2]
```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yu199195) [inline shenyu-agent/shenyu-agent-core/dependency-reduced-pom.xml:1]: remove this

File: shenyu-common/src/main/java/org/apache/shenyu/common/dto/convert/rule/impl/CryptorResponseRuleHandle.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    private String strategyName;
+
+    private String key;
+
+    private String fieldNames;
+
+    /**
+     * no constructor.
+     */
+    public CryptorResponseRuleHandle() {
```
Comment (yu199195) [inline shenyu-common/src/main/java/org/apache/shenyu/common/dto/convert/rule/impl/CryptorResponseRuleHandle.java:38]: remove this

File: shenyu-plugin/shenyu-plugin-cryptor/src/main/java/org/apache/shenyu/plugin/cryptor/common/config/SubscribeConfig.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Cryptor register config.
+ */
+public class SubscribeConfig {
```
Comment (yu199195) [inline shenyu-plugin/shenyu-plugin-cryptor/src/main/java/org/apache/shenyu/plugin/cryptor/common/config/SubscribeConfig.java:32]: why not use SPI Loader？

**Review by flycash — COMMENTED** [review]
and I think you could add integrated tests in this PR or create a new PR.
We'd better make sure that we have not only unit tests but also integrated tests for every new feature.

File: shenyu-agent/shenyu-agent-bootstrap/dependency-reduced-pom.xml (hunk 1 above)
Comment (yu199195) [reply shenyu-agent/shenyu-agent-bootstrap/dependency-reduced-pom.xml:2]: please romeve this

**Review by yu199195 — APPROVED** [review]

**Review by yu199195 — APPROVED** [review]

**Pull request merged**